File: src/api_server/logger.py
# logger.py
import os, time, json
LOG_VERBOSE = os.getenv("LOG_VERBOSE","true").lower()=='true'
def log(level, mod, trace, event, **fields):
    head = f"[{level}][{mod}][{trace}] {event}"
    # 使用 ensure_ascii=False 來正確處理中文字元
    body = ", ".join([f"{k}={json.dumps(v, ensure_ascii=False)}" for k,v in fields.items()])
    print(f"{head} {body}" if body else head, flush=True)

File: src/js/utils/logger.js
// src/js/utils/logger.js
// 根據使用者需求，LOG_VERBOSE 可由 window 物件或環境變數設定
export const LOG_VERBOSE = window.LOG_VERBOSE ?? true;

/**
 * 標準化日誌函式
 * @param {'INFO'|'WARN'|'ERROR'} level - 日誌等級
 * @param {string} mod - 模組名稱 (e.g., 'SpacePlanning', 'agentAPI')
 * @param {string} trace - 追蹤 ID (trace_id)
 * @param {string} event - 事件描述 (e.g., '準備送出打包請求')
 * @param {Object} [fields={}] - 附帶的結構化資料 (key-value pairs)
 */
export function log(level, mod, trace, event, fields = {}) {
  const head = `[${level}][${mod}][${trace}] ${event}`;
  const kv = Object.entries(fields)
    .map(([k, v]) => `${k}=${JSON.stringify(v)}`)
    .join(', ');
  
  const message = kv ? `${head} ${kv}` : head;

  switch (level) {
    case 'ERROR':
      console.error(message);
      break;
    case 'WARN':
      console.warn(message);
      break;
    default:
      console.log(message);
      break;
  }
}
